resolver/master_changer_test.go

```diff
@@ -0,0 +1,91 @@
+package resolver
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+func assertMastersState(mc *MasterChanger, want []string) error {
+	var got []string
+
+	select {
+	case got = <-mc.ChangedChan():
+	default:
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		return fmt.Errorf("got %q, want %q", got, want)
+	}
+
+	return nil
+}
+
+func TestMasterChanger_UpdatedMasters(t *testing.T) {
+	mc := NewMasterChanger([]string{"b:5050", "a:5050"}, 1)
+
+	for i, tt := range []struct {
+		masters, want []string
+	}{
+		{
+			[]string{"c", "d", "b"},
+			[]string{"", "b:5050", "c:5050", "d:5050"},
+		},
+		{
+			[]string{"e"},
+			[]string{"", "e:5050"},
+		},
+		{
+			[]string{},
+			nil,
+		},
+	} {
+		ms := make([]*mesos.MasterInfo, len(tt.masters))
+
+		for im, m := range tt.masters {
+			curm := m
+			ms[im] = &mesos.MasterInfo{Hostname: &curm}
+		}
+
+		fmt.Printf("%+v\n", ms)
+		mc.UpdatedMasters(ms)
+
+		if err := assertMastersState(mc, tt.want); err != nil {
+			t.Fatalf("test #%d: %v", i, err)
+		}
+	}
+}
+
+func TestMasterChanger_OnMasterChanged(t *testing.T) {
+	mc := NewMasterChanger([]string{"b:5050", "a:5050"}, 1)
+
+	for i, tt := range []struct {
+		master string
+		want   []string
+	}{
+		{
+			"a",
+			[]string{"a:5050", "b:5050"},
+		},
+		{
+			"c",
+			[]string{"c:5050", "b:5050"},
+		},
+		{
+			"b",
+			[]string{"b:5050"},
+		},
+		{
+			"",
+			[]string{"0.0.0.0:5050"},
+		},
+	} {
+		mc.OnMasterChanged(&mesos.MasterInfo{Hostname: &tt.master})
```

Comment:
Actually there is in line 81 (covering masterHostPort 100%), I'll add explicit comments